app/logic/users.py

```diff
@@ -10,28 +12,79 @@ def isEligibleForProgram(program, user):
     :param user: accepts a User object or userid
     :return: True if the user is not banned and meets the requirements, and False otherwise
     """
-
-    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program).exists()):
+    now = datetime.datetime.now()
+    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program, ProgramBan.endDate > now).exists()):
         return False
 
     return True
 
-def addRemoveInterest(rule, program_id, currentUser):
+def addUserInterest(program_id, username):
+    """
+    This function is used to add an interest to .
+    Parameters:
+    program_id: id of the program the user is interested in
+    username: username of the user showing interest
+    """
+    try:
+        Interest.get_or_create(program = program_id, user = username)
+        return "Successfully added interest"
+    except Exception as e:
+        print("Error: ", e)
+        return "An error ocurred while adding the interest"
+
+def removeUserInterest(program_id, username):
     """
     This function is used to add or remove interest from the interest table.
     Parameters:
-    rule: Gets the url from the ajax call, specifies to add or remove interest
     program_id: id of the program the user is interested in
+    username: username of the user showing disinterest
+
     """
+    interestToDelete = Interest.get(Interest.program == program_id, Interest.user == username)
+    if interestToDelete:
+        interestToDelete.delete_instance()
+        return "Successfully removed interest"
+    else:
+        raise Exception("Interest does not exist.")
+        return False
```

Comment:
You didn't address my earlier comment about this. This case shouldn't cause a failure. What they want the end result to be is already the case - they aren't interested in this particular program. Also, your `return False` will never be reached because you are raising the exception, but there's not a good reason to raise the exception in the first place.

Reply:
We have removed the unnecessary exception raised because if program_id and username does not except exceptions have already been taken care of in the routes.py file.